src/quacc/util/defects.py

```diff
@@ -0,0 +1,174 @@
+"""
+Utility functions for dealing with defects
+"""
+from __future__ import annotations
+
+import numpy as np
+from ase.atoms import Atoms
+from pymatgen.analysis.defects.core import Defect
+from pymatgen.analysis.defects.generators import (
+    AntiSiteGenerator,
+    ChargeInterstitialGenerator,
+    InterstitialGenerator,
+    SubstitutionGenerator,
+    VacancyGenerator,
+    VoronoiInterstitialGenerator,
+)
+from pymatgen.analysis.defects.thermo import DefectEntry
+from pymatgen.core import Structure
+from pymatgen.core.periodic_table import DummySpecies
+from pymatgen.entries.computed_entries import ComputedStructureEntry
+from pymatgen.io.ase import AseAtomsAdaptor
+from shakenbreak.input import Distortions
+
+# NOTES:
+# - Anytime an Atoms object is converted to a pmg structure, make sure
+# to reattach any .info flags to the Atoms object, e.g. via `new_atoms.info = atoms.info.copy()``.
+# Note that atoms.info is mutable, so copy it!
+# - All major functions should take in Atoms by default and return Atoms
+# by default. Pymatgen structures can be returned with an optional kwarg.
+# - If you modify the properties of an input Atoms object in any way, make sure to do so
+# on a copy because Atoms objects are mutable.
+
+
+def make_defects_from_bulk(
```

Comment:
I removed the `DummySpecies` as a user-supplied option. I think it's safe for us to assume that "X" should be used. Happy to revert if you strongly disagree (perhaps if there are multiple types of dummy species? Does that ever happen?).